src/java_bytecode/java_string_library_preprocess.h

```diff
@@ -52,7 +52,9 @@ class java_string_library_preprocesst:public messaget
   void add_string_type(const irep_idt &class_name, symbol_tablet &symbol_table);
   bool is_known_string_type(irep_idt class_name);
 
+#ifndef UNIT_TEST
```

Comment:
There are several options in decreasing preference:
1. Factor these methods out into a separate class and test them through its public interface.
2. Make the test class a friend (putting the friend declaration between #ifdef UNIT_TEST is a good idea)
3. Turn private into protected and test them through a derived class in your test.